enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+
+#### Future work
+
+There are several topics in this area that would be wise to defer for upcoming phases, namely:
+
+1. Define missing capabilities and their concrete behaviour, for example, add a "capability" for machine-api
+2. Precisely define the reaction of the operators listed below for the "External" platform type
+3. Define and document concrete mechanism for supplement cluster with provider-specific components during installation time (CCM, MAPI controller, DNS controller)
+
+### Risks and Mitigations
+
+#### Support burden
+There is concern that *random* customers will use this feature out of context
+and create a support burden. However, using such platform type in conjunction with [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+could give us a clear signal about how to properly triage 3d party created replacements for Red Hat components.
+
+### Drawbacks
+
+Current approach with having statically defined platform list is already there, quite transparent and battle-hardened.
+Changing it by adding the new, way less specific platform type would mean a significant step away from current design patterns in this area.
+
+Also, future support strategy is not completely clear due to we have plans around enablement of 3d party platform-specific components, which
+codebase would be mostly out of our control.
+
+## Design Details
+
+### Open Questions
+
+- Should we explicitly communicate that the "External" platform is something that we do not support yet?
+- Should we gate the "External" platform addition behind the feature gate by generating separate CRD for TPNU clusters?
+  - Do we already have all the necessary mechanisms for gating such API changes behind the feature gate?
+- Should we invest in preparing workflows that will perform UPI with the "None"/"External" platform types installation
+  on the AWS or GCP, or existed vSphere-based workflows would be enough?
+
+### Test Plan
+
+During the first stages, we must ensure that OpenShift [built-in operators whose behaviour depends on platform type](#affected-components) treat the "External" platform type the same way as "None".
+To achieve this - existing infrastructure and mechanisms employed for exercising topologies with the "None" platform type might be used.
+
+For the moment of writing, the only workflow where the "None" platform tests is [upi-conf-vsphere-platform-none-ovn](https://steps.ci.openshift.org/workflow/upi-vsphere-platform-none-ovn).
+Based on this workflow, a new one with platform type "External" with a respective set of jobs should be added to ensure that we do not disrupt the current openshift operation.
+
+However, given that vSphere is the only platform where we're exercising clusters installation with platform "None" specified,
+it would be beneficial to develop additional workflows using a provider with better capacity and performance (AWS or GCP, perhaps).
```

Comment:
Could we achieve similar coverage by using in-repo functional tests instead of building another e2e job?

Reply:
I'm not sure what you mean under "in-repo functional tests" here...
Unfortunately, there is more than one component involved, and we might need some top-level signal. 

Speaking about in-repo tests, yes, we could and should extend existing tests for each component, but i'm a bit concerned about orchestrating and controlling this, OCP is big.

Reply:
For example, it should be possible to write a test for the MCO to verify that when the platform is External it generates the right manifests, and we shouldn't need to deploy an entire cluster to verify that subset of the behavior. Can we perform similar tests for all of the other components, and not have full end-to-end and conformance tests of the External platform configuration?